core/iwasm/common/arch/invokeNative_riscv.S

```diff
@@ -0,0 +1,134 @@
+
+/*
+ * Arguments passed in:
+ *
+ * a0 function ptr
+ * a1 argv
+ * a2 nstacks
+ */
+
+/*
+ * sp (stack pointer)
+ *    |- sd/sw to store 64/32-bit values from register to memory
+ *    |- ld/lw to load from stack to register
+ * fp/s0 (frame pointer)
+ * a0-a7 (8 integer arguments)
+ *    |- sd/sw to store
+ *    |- ld/lw to load
+ * fa0-a7 (8 float arguments)
+ *    |- fsd/fsw to store
+ *    |- fld/fsw to load
+ * t0-t6 (temporaries regisgers)
+ *    |- caller saved
+ */
+
+        /* reserve space on stack to save return address and frame pointer */
+        addi             sp, sp, -16
+        RV_OP_STOREREG   fp, 0 * RV_REG_SIZE(sp)    /* save frame pointer */
+        RV_OP_STOREREG   ra, 1 * RV_REG_SIZE(sp)    /* save return address */
+
+        mv               fp, sp                     /* set frame pointer to bottom of fixed frame */
+
+        /* save function ptr, argv & nstacks */
+        mv               t0, a0                     /* t0 = function ptr */
+        mv               t1, a1                     /* t1 = argv array address */
+        mv               t2, a2                     /* t2 = nstack */
+
+#ifndef __riscv_float_abi_soft
+        /* fill in fa0-7 float-registers*/
+        RV_OP_LOADFPREG  fa0, 0 * RV_FPREG_SIZE(t1) /* fa0 = argv[0] */
+        RV_OP_LOADFPREG  fa1, 1 * RV_FPREG_SIZE(t1) /* fa1 = argv[0] */
+        RV_OP_LOADFPREG  fa2, 2 * RV_FPREG_SIZE(t1) /* fa2 = argv[0] */
+        RV_OP_LOADFPREG  fa3, 3 * RV_FPREG_SIZE(t1) /* fa3 = argv[0] */
+        RV_OP_LOADFPREG  fa4, 4 * RV_FPREG_SIZE(t1) /* fa4 = argv[0] */
+        RV_OP_LOADFPREG  fa5, 5 * RV_FPREG_SIZE(t1) /* fa5 = argv[0] */
+        RV_OP_LOADFPREG  fa6, 6 * RV_FPREG_SIZE(t1) /* fa6 = argv[0] */
+        RV_OP_LOADFPREG  fa7, 7 * RV_FPREG_SIZE(t1) /* fa7 = argv[0] */
+        addi             t1, t1, RV_FPREG_SIZE * 8
+#endif
+
+        /* fill in a0-7 integer-registers*/
+        RV_OP_LOADREG    a0, 0 * RV_REG_SIZE(t1)    /* a0 = argv[8] */
+        RV_OP_LOADREG    a1, 1 * RV_REG_SIZE(t1)    /* a1 = argv[9] */
+        RV_OP_LOADREG    a2, 2 * RV_REG_SIZE(t1)    /* a2 = argv[10] */
+        RV_OP_LOADREG    a3, 3 * RV_REG_SIZE(t1)    /* a3 = argv[11] */
+        RV_OP_LOADREG    a4, 4 * RV_REG_SIZE(t1)    /* a4 = argv[12] */
+        RV_OP_LOADREG    a5, 5 * RV_REG_SIZE(t1)    /* a5 = argv[13] */
+        RV_OP_LOADREG    a6, 6 * RV_REG_SIZE(t1)    /* a6 = argv[14] */
+        RV_OP_LOADREG    a7, 7 * RV_REG_SIZE(t1)    /* a7 = argv[15] */
+
+        addi             t1, t1, RV_REG_SIZE * 8    /* t1 points to stack args */
+
+        /* directly call the function if no args in stack,
+           x0 always holds 0 */
+        beq              t2, x0, call_func
+
+        /* reserve enough stack space for function arguments */
+        sll              t3, t2, RV_REG_SHIFT       /* shift left 3 bits. t3 = n_stacks * 8 */
+        sub              sp, sp, t3
+
+        /* make 16-byte aligned */
+        and              sp, sp, ~(15LL)
```

Comment:
For 32 bit, the original value here is ~15, not sure whether it will cause compile warning if using ~(15LL)?

Reply:
No error or warning on latest compiler (gcc 10) found.